my_pyscf/mcscf/lasscf_rdm.py

```diff
@@ -117,48 +118,60 @@ def get_init_guess_rdm (las, mo_coeff=None, h2eff_sub=None):
+        ''' "fcibox.kernel" should return e_cas, (casdm1rs, casdm2r) '''
+        def get_veff (my_casdm1frs):
+            casdm1fs = las.make_casdm1s_sub (casdm1frs=my_casdm1frs)
+            my_veff = las.get_veff (dm1s=las.make_rdm1 (mo_coeff=mo_coeff, casdm1s_sub=casdm1fs))
+            my_veff = las.split_veff (my_veff, h2eff_sub, mo_coeff=mo_coeff, casdm1s_sub=casdm1fs)
+            return my_veff
+        converged = False
         e_cas, fakeci = lasci_sync.ci_cycle (las, mo_coeff, None, veff, h2eff_sub, casdm1frs, log)
         casdm1frs = [f[0] for f in fakeci]
         casdm2fr = [f[1] for f in fakeci]
         veff = get_veff (casdm1frs)
         e_tot = las.energy_nuc () + las.energy_elec (mo_coeff=mo_coeff, h2eff=h2eff_sub,
+                                                 casdm1frs=casdm1frs, casdm2fr=casdm2fr)
+        log.info ("LASSCF rdm-jk init : E = %.15g", e_tot)
+        for it in range (max_cycle_rdmjk):
+            casdm1frs_old = casdm1frs
+            casdm2fr_old = casdm2fr
+            e_old = e_tot
+            e_cas, fakeci = lasci_sync.ci_cycle (las, mo_coeff, None, veff, h2eff_sub, casdm1frs, log)
+            casdm1frs = [f[0] for f in fakeci]
+            casdm2fr = [f[1] for f in fakeci]
+            veff = get_veff (casdm1frs)
+            e_tot = las.energy_nuc () + las.energy_elec (mo_coeff=mo_coeff, h2eff=h2eff_sub,
                                                      casdm1frs=casdm1frs, casdm2fr=casdm2fr)
-        ddm = np.concatenate ([(d1 - d0).ravel () for d1, d0 in zip (casdm1frs, casdm1frs_old)])
-        ddm = np.append (ddm, np.concatenate (
-            [(d1 - d0).ravel () for d1, d0 in zip (casdm2fr, casdm2fr_old)]
-        ))
-        ddm = linalg.norm (ddm)
-        log.info ("LASSCF rdm-jk %d : E = %.15g ; dE = %.15g ; |ddm| = %.15g", it+1, e_tot,
-                  e_tot-e_old, ddm)
-        if (ddm < conv_tol_rdmjkddm) and (abs(e_tot-e_old) < conv_tol_rdmjkde):
-            converged = True
-            break
-    log.info ("LASSCF rdm-jk {}".format (('not converged','converged')[int(converged)]))
-    return e_cas, casdm1frs, casdm2fr, converged
+            ddm = np.concatenate ([(d1 - d0).ravel () for d1, d0 in zip (casdm1frs, casdm1frs_old)])
+            ddm = np.append (ddm, np.concatenate (
+                [(d1 - d0).ravel () for d1, d0 in zip (casdm2fr, casdm2fr_old)]
+            ))
+            ddm = linalg.norm (ddm)
+            log.info ("LASSCF rdm-jk %d : E = %.15g ; dE = %.15g ; |ddm| = %.15g", it+1, e_tot,
+                      e_tot-e_old, ddm)
+            if (ddm < conv_tol_rdmjkddm) and (abs(e_tot-e_old) < conv_tol_rdmjkde):
+                converged = True
+                break
+        log.info ("LASSCF rdm-jk {}".format (('not converged','converged')[int(converged)]))
+        return e_cas, casdm1frs, casdm2fr, converged
 
 def kernel (las, mo_coeff=None, casdm1frs=None, casdm2fr=None, conv_tol_grad=1e-4, verbose=lib.logger.NOTE):
```

Comment:
Editing this function changes the behavior of the existing lasscf_rdm implementation as well as your new implementation. Can you instead write a separate kernel function for your subclass, if you must change something in kernel?

Reply:
Or, failing that, you could instead write "hook" functions that do nothing for the lasscf_rdm.LASSCFNoSymm class but which do the I/O for the lasscf_rdm.extremeAsynLASSCF class.